test/config.jl

```diff
@@ -0,0 +1,39 @@
+using Test
+import ClimaAtmos as CA
+
+include(joinpath("..", "perf", "common.jl"))
+
+@testset "AtmosCoveragePerfConfig" begin
+    # Just default config overridden by `default_perf.yml`
+    config = AtmosCoveragePerfConfig()
+    # Test defaults overridden by `default_perf`
+    @test config.parsed_args["dt"] == "1secs"
+    # Test defaults not overridden by `default_perf`
+    @test config.parsed_args["y_elem"] == 6
+
+    # Test with `target_job`
+    config = AtmosCoveragePerfConfig(;
+        config_dict = CA.config_from_target_job("sphere_baroclinic_wave_rhoe"),
+    )
+    # Target job config overridden by `default_perf`
+    @test config.parsed_args["dt"] == "1secs"
```

Comment:
Shouldn't `sphere_baroclinic_wave_rhoe` have the precedence over the default (as per docstring), so that `dt != "1secs"`?

Reply:
Sorry, the docstring is incorrect - the default perf config needs to take priority to keep existing performance tests consistent.

Priority:
1. Default perf configuration (to increase coverage)
2. Configuration from the given config file/dict
3. Default configuration

Reply:
If the target_job is overridden, what is its use? 
```
    config = AtmosCoveragePerfConfig(;
        config_dict = CA.config_from_target_job("sphere_baroclinic_wave_rhoe"),
    )
```
Is for a moist configuration with vertical diffiusion, Held Suarzed forcing, precipitation, radiation.

The only meaningful configuration that is left from `sphere_baroclinic_wave_rhoe` is that the initial condition is `DryBaroclinicWave`.

Reply:
If a user wants to test the performance of a specific job, they can use `TargetJobConfig`.
`AtmosCoveragePerfConfig` is essentially only used to create configurations for the existing performance tests.

Reply:
Given that, what I would recommend to reframe `AtmosCoveragePerfConfig` as being: simulate everything, optionally add more configuration using the `config_dict` (even if it is very unusual that a user provided option is ignored in favor of a default one). Then, we remove all the references to target_jobs in `AtmosCoveragePerfConfig` and only add those options that we care about for our tests. 

My point is that the notion of target_job attached in any way to an `AtmosCoveragePerfConfig` that enables all the physics is really confusing and counter intuitive. I think that my proposed change would make the intent clear: instead of "targeting a edmf job" we "add emdf" to our configuration.

Reply:
I agree that the current way we use `AtmosCoveragePerfConfig` is not intuitive to users, but it is needed to maintain our tests. This should be changed, but I think that this would need to be part of a larger discussion of our performance testing practices, since it would break continuity on said tests.

Reply:
Yes, what I am saying is that we maintain our tests by explicitly adding those options that we need.

For example, if we had a `sphere_baroclinic_rhoe` test, we would just need to add `initial_conditions = DryBaroclinicWave`.